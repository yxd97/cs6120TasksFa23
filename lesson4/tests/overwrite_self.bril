@main {
  a: int = const 1;
  b: int = const 2;
  print a;
  a: int = add a b;
  print a;
}
