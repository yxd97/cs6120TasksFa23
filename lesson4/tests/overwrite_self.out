@main {
  a: int = const 1; # defs: a_DEF1, 
  b: int = const 2; # defs: b_DEF1, 
  print a;          # uses: USE1,                # reaching defs: a_DEF1, b_DEF1, 
  a: int = add a b; # defs: a_DEF2, uses: USE2,  # reaching defs: a_DEF1, b_DEF1, 
  print a;          # uses: USE3,                # reaching defs: b_DEF1, a_DEF2, 
}
